packages/apps/human-app/frontend/src/modules/worker/jobs/services/jobs.service.ts

```diff
@@ -0,0 +1,155 @@
+  type UiConfig,
+} from '../types';
+
+const apiPaths = {
+  jobs: '/jobs',
+  myJobs: '/assignment/job',
+  assignJob: '/assignment/job',
+  resignJob: '/assignment/resign-job',
+  refreshJobs: '/assignment/refresh',
+  uiConfig: '/ui-config',
+};
+
+export class JobsService {
+  private readonly authorizedHttpApiClient: AuthorizedHttpApiClient;
+
+  constructor() {
+    const httpClient = new HttpApiClient(env.VITE_API_URL);
+    const authService = new AuthService(httpClient);
+    this.authorizedHttpApiClient = new AuthorizedHttpApiClient(
+      env.VITE_API_URL,
+      authService
+    );
+  }
+
+  async fetchAvailableJobs(args: JobsBody) {
+    try {
+      const result =
+        await this.authorizedHttpApiClient.get<AvailableJobsSuccessResponse>(
+          apiPaths.jobs,
+          {
+            queryParams: args.queryParams,
+            successSchema: availableJobsSuccessResponseSchema,
+            abortSignal: args.signal,
+          }
+        );
+
+      return result;
+    } catch (error: unknown) {
+      if (error instanceof ApiClientError) {
+        throw error;
+      }
+      throw new Error('Failed to get available jobs');
+    }
+  }
+
+  async fetchMyJobs(args: JobsBody) {
+    try {
+      const result =
+        await this.authorizedHttpApiClient.get<MyJobPaginationResponse>(
+          apiPaths.myJobs,
+          {
+            queryParams: args.queryParams,
+            successSchema: myJobsSuccessResponseSchema,
+            abortSignal: args.signal,
+          }
+        );
+
+      return result;
+    } catch (error: unknown) {
+      if (error instanceof ApiClientError) {
+        throw error;
+      }
+      throw new Error('Failed to get my jobs');
+    }
+  }
+
+  async assignJob(data: AssignJobBody) {
+    try {
+      const result = await this.authorizedHttpApiClient.post(
+        apiPaths.assignJob,
+        {
+          body: { ...data },
+        }
+      );
+
+      return result;
+    } catch (error: unknown) {
+      if (error instanceof ApiClientError) {
+        throw error;
+      }
+      throw new Error('Failed to assign job');
+    }
+  }
+
+  async resignJob(data: RejectTaskBody) {
+    try {
+      const result = await this.authorizedHttpApiClient.post(
+        apiPaths.resignJob,
+        {
+          body: { ...data },
+        }
+      );
+
+      return result;
+    } catch (error: unknown) {
+      if (error instanceof ApiClientError) {
+        throw error;
+      }
+      throw new Error('Failed to resign job');
+    }
+  }
+
+  async refreshJobs(data: RefreshJobsBody) {
+    try {
+      const result = await this.authorizedHttpApiClient.put(
+        apiPaths.refreshJobs,
+        {
+          body: { ...data },
+        }
+      );
+
+      return result;
+    } catch (error: unknown) {
+      if (error instanceof ApiClientError) {
+        throw error;
+      }
+      throw new Error('Failed to refresh jobs');
+    }
+  }
+
+  async getUiConfig() {
```

Comment:
It shouldn't be part of the jobs service. Right now it has only `chainIdsEnabled`, but in general it can have some other params used here and there, so should be general